scripts/benchmark/checks/deploy_dos.py

```diff
@@ -0,0 +1,312 @@
+import logging
+import pathlib
+from typing import (
+    Tuple
+)
+
+from web3 import (
+    Web3
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+)
+
+from eth.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from eth.chains.base import (
+    MiningChain,
+)
+from eth.rlp.blocks import (
+    BaseBlock,
+)
+
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+)
+from utils.compile import (
+    get_compiled_contract
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+from abc import (
+    abstractmethod
+)
+
+FIRST_TX_GAS_LIMIT = 367724
+SECOND_TX_GAS_LIMIT = 62050
+THIRD_TX_GAS_LIMIT = 104789
+FORTH_TX_GAS_LIMIT = 21272
+FIFTH_TX_GAS_LIMIT = 21272
+
+CONTRACT_FILE = 'scripts/benchmark/contract_data/DOSContract.sol'
+CONTRACT_NAME = 'DOSContract'
+
+W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
+
+
+class BaseDOSContractBenchmark(BaseBenchmark):
+
+    def __init__(self, num_blocks: int = 100, num_tx: int = 2) -> None:
```

Comment:
Depending on the time these benchmarks take maybe we could bump up the transactions to really *fill* the blocks. I know that's a bit of a balance act because the benchmarks still need to finish timely to not slow down CI too much. That said, maybe we want to shift towards *less* blocks but with *more* transactions so that the benchmark is less impacted by the general block forging costs and more by the actual costs of transaction processing?

Maybe have @pipermerriam weigh in on that.

Reply:
I increased the TX per block to 10 and decrease the number of blocks to 20. It now takes around 0.6-0.8 seconds longer to build all the 200 TXs.